app/models/comment.rb

```diff
@@ -49,14 +51,24 @@ class Comment < ActiveRecord::Base
     participation.unparticipate! if participation.present?
   end
 
-  def message
-    @message ||= Diaspora::MessageRenderer.new text
-  end
-
   def text= text
      self[:text] = text.to_s.strip #to_s if for nil, for whatever reason
   end
 
+  def people_allowed_to_be_mentioned
+    @people_allowed_to_be_mentioned ||= if parent.public?
```

Comment:
It is called twice on each comment generation, but in different workers, so caching doesn't help.